### PR TITLE
Draft - 2420 -changes for migrating BlockHeader api out of access API

### PR DESCRIPTION
**Pull request by koli19761**
Initial changes for migrating BlockHeader api out of access API

### CODE REVIEW

File: access/legacy/handler.go [hunk 1]
```diff
@@ -15,12 +15,14 @@ import (
 
 type Handler struct {
 	api   access.API
+	psapi access.PROTOCOL_STATE_API
```
Comment (koko1123) [inline access/legacy/handler.go:18]: You dont' need to change the legacy handler! this is only for the new "observer"! https://github.com/onflow/flow-go/pull/2370/files#diff-c396d8ee377761a43fca284545cb4ee535f868ea2b16db01db81a21741855ed8R1-R28

**Review by koko1123 — CHANGES_REQUESTED** [review]
I'm not sure I've explained this correctly, these changes belong in the new engine!

File: engine/access/rest/accounts.go [hunk 2]
```diff
@@ -7,15 +7,15 @@ import (
 )
 
 // GetAccount handler retrieves account by address and returns the response
-func GetAccount(r *request.Request, backend access.API, link models.LinkGenerator) (interface{}, error) {
+func GetAccount(r *request.Request, backend access.API, psapi access.PROTOCOL_STATE_API, link models.LinkGenerator) (interface{}, error) {
```
[the remark this answers is not in the document]
Comment (koli19761) [reply engine/access/rest/accounts.go:10]: Ok , i will modify that

File: access/legacy/handler.go (hunk 1 above)
Comment (koli19761) [reply access/legacy/handler.go:18]: Sure will make the change

**Pull request closed without merging**